src/oscar/views/generic.py

```diff
@@ -27,7 +27,15 @@ def post(self, request, *args, **kwargs):
                 return self.response
             else:
                 messages.error(request, _("Invalid form submission"))
-        return super().post(request, *args, **kwargs)
+                return self.get()
+
+        try:
+            f = super().post
```

Comment:
We can try to directly `return` here rather than setting `f` since the except block is also returning.

Reply:
That would potentially catch exception from inside the `post(..)`, and add an incorrect message, but done anyway as this is obscure, and doubtless is dev errors anyway not seen by users.